### DISCUSSION TITLE
Add additional Airspy R2 sampling rates.

### DISCUSSION THREAD

**Discussion by pmaine** [AlexandreRouma/SDRPlusPlus#335] [Q&A] [answered]
My Airspy R2 plugged into an Ubuntu laptop with a 2.5Ghz Intel I5 processor is able run in SDR++ at 10MSPS.

When I run the Airspy command line tool on my Pi 4B I get just under 8MSPS.

SDR++ currently provides a 2.5 and 10MSPS.

How can I add 3,4,5,6,7, 7.5?

Thank you

**Comment by AlexandreRouma** [comment, marked as answer]
You can't, those are the only two sample rates supported by the Airspy R2.

You could try decimation which will allow you to divide the sample rate by 2, 4, 8, 16, etc... but this might use more CPU than with no decimation since it has to be done in software.

**Reply by pmaine** [reply]
Can I change decimation on SDR++ GUI?

**Reply by AlexandreRouma** [reply]
Yes, last setting in the source menu.

**Comment by jturowsk** [comment]
I visited a webside from dd1us (german radio amateur) and he said: by changing the **device_sample_rate=6000000** in spyserver.config it is possible to run the airspy R2 with 6 MSPS. So I ran **airspy_rx -r NUL -a 6000000** on Linux and it says _Streeaming at 5.997 MSPS_ and so further on. It looks like this sample_rate is supported, but not official. This sample rate could be perfect to receive HRPT images, because 2.5 MSPS is to low and 10 MSPS is not really stable enough. 
By the way, perhaps it would be nice if the sample rate in sdr++ could be changed manually as an additional option.

**Reply by AlexandreRouma** [reply]
I don't care what manual hacks can do. If it was supported properly, Airspy would list it as an option. SDR++ doesn't have a hardcoded list of options, it asks the driver what the device supports.

**Comment by jturowsk** [comment]
Thanks for your quick answer. The possibility to do so is not a hack, it's in the latest original firmware and it's working.
You are write, you wouldn't get the answer, if you ask the stick for supported sample rates and when you do so in the sdr++ then 
this sample rate will not show up. But an option for a manually input is still for discussion, unless it doesn't fit your overall concept
of sdr++. By the way this software ist still great for me, especially on Linux.
